src/sentry/static/sentry/app/views/performance/data.tsx

```diff
@@ -283,7 +302,9 @@ function generateGenericPerformanceEventView(
   }
   savedQuery.query = stringifyQueryObject(conditions);
 
-  return EventView.fromNewQueryWithLocation(savedQuery, location);
+  const eventView = EventView.fromNewQueryWithLocation(savedQuery, location);
+  eventView.addOverrideCondition('event.type', ['transaction']);
```

Comment:
I wonder if this is the right interface for this. Would it make sense to allow for `overrideCondition` to be added directly to the `savedQuery` object and the `eventView` will handle the rest?

Reply:
Yeah that's a good call, I don't think the extra indirection is necessary.